src/devices_models/devices/common/range_constraint.jl

```diff
@@ -107,7 +107,7 @@ function device_semicontinuousrange(
 )
     time_steps = model_time_steps(psi_container)
     varcts = get_variable(psi_container, inputs.variable_name)
-    varbin = get_variable(psi_container, inputs.bin_variable_name)
+    varbin = get_variable(psi_container, inputs.bin_variable_name[1])
```

Comment:
Can you add asserts for the expected sizes of inputs.bin_variable_name in all functions?